docs/dev-process/coding-guidelines.md

```diff
@@ -1,92 +1,179 @@
-
-# C# Coding Style
+# C# Coding Guidelines
 
 ## Coding Conventions
 
 As a general rule, our coding convention is to follow the style of the surrounding code.
-Avoid reformatting any code when submitting a PR as it obscures the functional changes of your change.
-We run the [.NET code formatter tool](https://github.com/dotnet/codeformatter) regularly help keep consistent formatting.
+So if a file happens to differ in style from conventions defined here
+(e.g. private members are named `m_member` rather than `_member`),
+the existing style in that file takes precedence.
+
+When making changes, you may find some existing code goes against the conventions defined here.
+In such cases, please avoid reformatting any existing code when submitting a PR as it obscures the functional changes of the PR.
+A separate PR should be submitted for style-only changes.
+We also run the [.NET code formatter tool](https://github.com/dotnet/codeformatter) regularly to keep consistent formatting.
+
+### Naming Conventions
+
+* Use meaningful, descriptive words for names.
+  For method names, it's encouraged to use `Verb-Object` pair such as **`LoadModule`**.
+
+* Use `_camelCase` to name internal and private fields and use `readonly` where possible.
+  Prefix instance fields with `_`, static fields with `s_` and thread static fields with `t_`.
+  When used on static fields, `readonly` should come after `static` (i.e. `static readonly` not `readonly static`).
```

Comment:
this is generally a linter job to enforce such things.
We have a hand-written linter for tabs vs spaces and such.
Do we consider to add a proper dotnet linter to CI and build.psm1?

Reply:
@TravisEz13 has some ideas to enforce the formatting check in CI.